### DISCUSSION TITLE
request 响应拦截里面处理了错误就不会走errorHandler？

### DISCUSSION THREAD

**Discussion by Cooper1886** [umijs/umi#11223] [Q&A]
如何在umi4项目里面，通过errorHandler处理完错误之后返回一个自定义格式的错误信息（code，message......）
我的需求是想要在页面queryInfo().catch(err => {})这里对错误做一些特别的操作，目前这里只能拿到axiosError。
umi3用umi-request的extend配置的请求，在errorhandler中可以return Promise.reject(err)，然后在页面请求中catch到。
补充：项目中后端并没有对错误做过多处理，好多都是直接浏览器报错。

**Comment by fz6m** [comment]
不推荐使用 umi request ，现代项目更推荐用 axios / fetch 做底层请求，自己配置响应拦截器，顶层用 [react-query](https://umijs.org/docs/max/react-query) 管理请求数据流，用 [valtio](https://umijs.org/docs/max/valtio) / [jotai](https://github.com/pmndrs/jotai) 管理全局数据流。

**Reply by Cooper1886** [reply]
意思是umi4没用umi-request了，只是提供了底层请求，需要自己配置react-query吗
